Turtles!

KeithCissell opened the pull request:
## Overview:
### Created two new enemies: Turtle and FlyingTurtle
Turtle
- Turtle walks around the map
- Abides to the laws of gravity
- If the player runs into it, they take damage
- When the turtle is stepped on, it hides in its shell
- When the turtle is in its shell:
	- Movement speed is initialized to 0
	- Player can stand on top of shell
	- Player can 'kick' shell
	- The shell's velocity is a major increase form walking
	- Moving shell will kill other enemies that it hits
FlyingTurtle
- Flaps its wings to glide accross the map
- The Y direction acceleration is decreased
- If player jumps on the FlyingTurtle's head, it looses its wings and turns into a normal Turtle

### Other Changes
Player now gets a verticle boost (bounce) into the air when he squashes enimies


## Testing:
### Running
To get the turtle in the map for testing, you need to add it in the playState.
- declare and add a Turtle and/or FlyingTurtle
- add `FlxG.overlap(<enemy>, <turtle>, <turtle>.enemyHitResolve);` with any enemy on the map
- add `FlxG.overlap(<player>, <turtle>, <turtle>.playerHitResolve);` with the player(s)
- add `FlxG.collide(<turtle>, <object>);` for any map, block, sentry, etc. that the turtle should not move through
- launch the game

### What to look for
- Both
	- Turtles move back and forth, turning around whenever they run into an object (wall, enemy, etc.)
	- Kills player when the enemy touches it from anywhere other than above
- Flying Turtle
	- Turtle maintains its height while flying. It will "flap" its wings after it falls to a certain hieght
	- Turtle's wings disappear when hit from above and will fall to the ground (turns into a normal turtle)
- Turtle
	- Hides in shell when hit from above by enemy
- Shelled Turtle
	- The shell falls to the ground without any X velocity
	- The shell will have no interaction with other enimies while static
	- The player will not die from touching the shell
	- If the player hits the shell from the side, the shell is launched in away from the player
	- Moving shell will kill any enemies it collides with
	- Shell will go back to 0 X velocity if the player jumps on top of it
- Other
	- Player will bounce into the air after jumping on top of a turtle or dontTouchMe


Reviewer: Dillon

KeithCissell committed:
Setting up work for making Turtle enemy

KeithCissell committed:
Finished initial set up of the Turtle class. Created graphics for the turtle and shell.

KeithCissell committed:
Turtle hides and shell. Player can kick shell around. Changed hitbox logic.

KeithCissell committed:
Turtle shell now kills enemies when kicked by player

KeithCissell committed:
Merged upstream 'master' into turtle_enemies

KeithCissell committed:
Created a flying turtle

KeithCissell committed:
Completed work on the Turtles and their interaction with the player

Dwooll94 reviewed (changes requested):
Documentation is all good, except for your testing plan allows the non-flying turtle to fall through the floor.

Also, in your play state I had to remove player collisions with the galoomba to get it to compile.

KeithCissell commented:
I updated the PlayState to use to proper hit resolve function for the turtle logic so that it compiles now.
Updated testing to add any collision logic with the map and any blocks in the PlayState.

KeithCissell committed:
Changed the method call for player-turtle overlap resolution.
